src/util/metrics/fd_metrics.h

```diff
@@ -0,0 +1,79 @@
+#ifndef HEADER_fd_src_util_metrics_fd_metrics_h
+#define HEADER_fd_src_util_metrics_fd_metrics_h
+
+#include "../fd_util_base.h"
+#include "../../tango/mcache/fd_mcache.h"
+#include <stdbool.h>
+
+struct metrics_t;
+typedef struct metrics_t metrics_t;
+
+typedef enum {
+  metrics_quic,
+  metrics_verify,
+  metrics_dedup,
+  metrics_pack,
+  metrics_bank,
+} metrics_tile_t;
+
+extern char * metrics_tile_names[];
+
+typedef enum {
+  datatype_float,
+  datatype_int,
+  datatype_string,
+  datatype_bool,
+} metrics_datapoint_t;
+
+#define METRICS_VALUE_STRING_SZ 128
+typedef struct {
+  long   i;
+  double f;
+  bool   b;
+  char   s[METRICS_VALUE_STRING_SZ];
+} metrics_value;
+
+#define METRICS_NAME_SZ 40
+
+typedef struct {
+  metrics_datapoint_t type;
+  char                name[METRICS_NAME_SZ];
+} Measurement;
+
+typedef struct {
+  Measurement            measurement;
+  metrics_value          value;
+} Datapoint;
+
+void
+metrics_boot( uchar const * pod, const metrics_tile_t tile, const ulong idx );
```

Comment:
The code in util/ has much more requirements than the rest of the codebase, since it's a library.

Could we follow the style in fd_log.h a lot more closely? Eg, these exported functions and types should be prefixed with `fd_` so importers don't get symbol collisions. Use FD_PROTOTYPES_BEGIN, I don't think fd_metrics_format should be exposed, or the enum etc?

As far as I can tell, fd_metrics.c could have an API that's basically like,

* fd_metrics_boot()`
* #FD_METRICS_ULONG(tag, value) ... (or fd_metrics_push).

Reply:
Move out of `util/` and now the prototype now looks like
```
void
fd_metrics_push( uint tag, ulong value )